### PR TITLE
[SPARK-49585][CONNECT] Replace executions map in SessionHolder with operationID set

### PR DESCRIPTION
**Pull request by ghost**
### What changes were proposed in this pull request?

SessionHolder has no reason to store ExecuteHolder directly as SparkConnectExecutionManager has a global map of ExecuteHolder. This PR replaces the map in SessionHolder with a set of operation IDs which is only used when interrupting executions within the session.

### Why are the changes needed?

Save memory, and simplify the code by making SparkConnectExecutionManager the single source of ExecuteHolders.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Existing tests.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Comment by changgyoopark-db** [comment]
HI again @juliuszsompolski , can you please review this change? Thanks a lot!

**Comment by changgyoopark-db** [comment]
Hi @hvanhovell Can you please review this change? Thanks!

File: sql/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SessionHolder.scala [hunk 1]
```diff
@@ -138,39 +137,38 @@ case class SessionHolder(userId: String, sessionId: String, session: SparkSessio
...
         messageParameters = Map("handle" -> sessionId))
     }
 
-    val oldExecute = executions.putIfAbsent(executeHolder.operationId, executeHolder)
-    if (oldExecute != null) {
-      // the existence of this should alrady be checked by SparkConnectExecutionManager
-      throw new IllegalStateException(
-        s"ExecuteHolder with opId=${executeHolder.operationId} already exists!")
+    var existed = true
+    operationIds.computeIfAbsent(
```
Comment (hvanhovell) [inline sql/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SessionHolder.scala:153]: How about this:
```scala
val alreadyExists = operationIds.putIfAbsent(operationId, ()) != null
```
That IMO is simpler and more readable.

Comment (hvanhovell) [reply sql/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SessionHolder.scala:153]: (basically the old code)

Comment (changgyoopark-db) [reply sql/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SessionHolder.scala:153]: Sadly, the compiler prohibits comparison between `Unit` and `null`; + Option[Unit](null) gives `Some(())`.

**Comment by changgyoopark-db** [comment]
@hvanhovell Hi! Can you please review and merge this PR? Thanks!

File: sql/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SessionHolder.scala (hunk 1 above)
Comment (juliuszsompolski) [reply sql/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SessionHolder.scala]: would `val alreadyExists = operationIds.putIfAbsent(operationId, ()) != Some(())`  work?

Comment (changgyoopark-db) [reply sql/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SessionHolder.scala]: Good idea, but unfortunately, doesn't work: warning.

[error] /Users/changgyoo.park/spark/sql/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SessionHolder.scala:153:67: comparing values of types Unit and Some[Unit] using `==` will always yield false

**Review by juliuszsompolski — APPROVED** [review]
LGTM. Thanks for the simplification!

**Comment by changgyoopark-db** [comment]
@hvanhovell @HyukjinKwon Good day! Could any of you please merge this trivial change? Thanks a lot!

**Review by allisonwang-db — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**